test/test_shared.cpp

```diff
@@ -2381,6 +2381,26 @@ TEST(Shared_MixedWithNonShared)
 #endif
 }
 
+
+#if REALM_ENABLE_ENCRYPTION
+// verify that even though different threads share the same encrypted pages,
+// a thread will not get access without the key.
+TEST(Shared_EncryptionKeyCheck)
```

Comment:
Better. But we still miss a test of the case where we try to open a SharedGroup with a crypt_key that is different from the original one. It should be easy to squeeze that into the first test.